server/api/src/rsGeneralAdmin.cpp

```diff
@@ -52,6 +53,51 @@ namespace
         return std::find_if(b, e, [](unsigned char _ch) { return ::isspace(_ch); }) != e;
     } // contains_whitespace
 
+     auto throw_if_user_type_can_result_in_non_functional_irods_server(
+        RsComm& rsComm,
+        const std::string_view _option,
+        const std::string_view _user_name,
+        const std::string_view _new_user_type
+    ) -> void {
+         if ("rodsadmin" == _new_user_type || "type" != _option) {
+             return;
+         }
+
+        BytesBuf* bbuf = nullptr;
+        irods::at_scope_exit free_buf { [&bbuf] { std::free(bbuf); }  };
+
+        if (const auto ec = rsZoneReport(&rsComm, &bbuf); ec < 0 ) {
+            const auto msg = "The server has experienced an unexpected error while processing the request. Please "
+                             "report to the administrator.";
+            addRErrorMsg(&rsComm.rError, SYS_NOT_ALLOWED, msg);
```

Comment:
`SYS_NOT_ALLOWED` -> `ec`

Should the message we return be more specific about what happened? After all, we are in `rsGeneralAdmin`, so the administrator is (hopefully) the one who is going to see the message anyway, right?

Reply:
That is a very good point.

Perhaps the error message becomes ... `"Failed to gather rodsadmin users managing a server in the local zone."`